Sources/SwiftCliCore/InputValidator.swift

```diff
@@ -11,140 +24,175 @@ public class InputValidator {
+        for (index, row) in pattern.enumerated() {
+            if row.isEmpty {
+                results.append(.failure(InputError.emptyRow(row: index + 1)))
+                anyerrors = true
+            } else {
+                continue
+            }
+        }
+        guard !anyerrors else {
+            return(results)
+        }
+        results.append(.success(Success.patternArray(pattern)))
+        return(results)
+    }
+
+    private func checkNoInvalidStitchesInNestedArray(pattern: [[String]]) -> Result<Success, InputError> {
+        let isEveryStitchValid = validateEachStitchInWholePattern(pattern: pattern)
+        switch isEveryStitchValid {
         case .success:
-            continue
+            return .success(Success.patternNestedArray(pattern))
         case .failure(let error):
-            throw error
+            return .failure(error)
         }
-    }
-    return pattern
-}
 
-private func checkNoInvalidStitchesInNestedArray(pattern: [[String]]) throws -> [[String]] {
-    let isEveryStitchValid = validateEachStitchInWholePattern(pattern: pattern)
-    switch isEveryStitchValid {
-    case .success:
-        return pattern
-    case .failure(let error):
-        throw error
     }
 
-}
-
-private func checkNoMathematicalIssuesInArrayOfRowInfo(pattern: [RowInfo]) throws -> [RowInfo] {
-    let isPatternMathematicallySound = validateEachRowWidth(allRowsMetaData: pattern)
-    switch isPatternMathematicallySound {
-    case .success:
-        return pattern
-    case .failure(let error):
-        throw error
+    private func checkNoMathematicalIssuesInArrayOfRowInfo(pattern: [RowInfo]) -> Result<Success, InputError> {
+        let isPatternMathematicallySound = validateEachRowWidth(allRowsMetaData: pattern)
+        switch isPatternMathematicallySound {
+        case .success:
+            return .success(Success.patternRowInfo(pattern))
+        case .failure(let error):
+            return .failure(error)
+        }
     }
-}
 
-private func validateEachStitch(stitchRow: [String], rowIndex: Int) -> Result<[String], InputError> {
+    private func validateEachStitch(stitchRow: [String], rowIndex: Int) -> Result<[String], InputError> {
 
-    var errorArray: [InputError] = []
-    for (index, stitch) in stitchRow.enumerated() {
-        if !isStitchValid(stitch: stitch) {
-            errorArray.append(InputError.invalidStitch(invalidStitch: stitchRow[index], rowLocation: index + 1))
+        var errorArray: [InputError] = []
+        for (index, stitch) in stitchRow.enumerated() {
+            if !isStitchValid(stitch: stitch) {
+                errorArray.append(InputError.invalidStitch(invalidStitch: stitchRow[index], rowLocation: index + 1))
+            }
         }
-    }
 
-    if errorArray.count > 0 {
-        return .failure(InputError.multipleErrors(errors: errorArray))
+        if errorArray.count > 0 {
+            return .failure(InputError.multipleErrors(errors: errorArray))
```

Comment:
I think it would make more sense to return a single error, `InputError.invalidStitch`, but instead of having `invalidStitch` and `rowLocation` as error parameters, use a dictionary instead, where the key is the location and the value is the invalid stitch.